src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -0,0 +1,271 @@
+                searchPreferences.isCaseSensitive());
+        caseSensitive.setToolTipText(Localization.lang("Case sensitive"));
+        caseSensitive.addActionListener(e -> {
+            searchPreferences.setCaseSensitive(caseSensitive.isSelected());
+            performSearch();
+        });
+
+        updateSearchModeButtonText();
+        searchModeButton.addActionListener(l -> {
+            int nextSearchMode = (searchDisplayMode.ordinal() + 1) % SearchDisplayMode.values().length;
+            searchDisplayMode = SearchDisplayMode.values()[nextSearchMode];
+            searchPreferences.setSearchMode(searchDisplayMode);
+            updateSearchModeButtonText();
+            performSearch();
+        });
+
+        clearSearchButton.setToolTipText(Localization.lang("Clear"));
+        clearSearchButton.addActionListener(l -> endSearch());
+
+        searchField.addFocusListener(Globals.focusListener);
+        searchField.addActionListener(e -> performSearch());
+        JTextFieldChangeListenerUtil.addChangeListener(searchField, e -> performSearch());
+        searchField.addKeyListener(new KeyAdapter() {
+            @Override
+            public void keyReleased(KeyEvent e) {
+                if (e.getExtendedKeyCode() == KeyEvent.VK_ESCAPE) {
+                    endSearch();
+                }
+            }
+        });
+
+        autoCompleteSupport.install();
+
+        setLayout(new WrapLayout(FlowLayout.RIGHT));
+        toolBar.setFloatable(false);
+        toolBar.add(currentResults);
+        toolBar.add(searchIcon);
+        toolBar.add(searchField);
+        toolBar.add(clearSearchButton);
+        toolBar.addSeparator();
+        toolBar.add(regularExp);
+        toolBar.add(caseSensitive);
+        toolBar.addSeparator();
+        toolBar.add(searchModeButton);
+        toolBar.addSeparator();
+        toolBar.add(new HelpAction(HelpFiles.SEARCH));
+        this.add(toolBar);
+    }
+
+    private void updateSearchModeButtonText() {
+        searchModeButton.setText(searchDisplayMode.getDisplayName());
+        searchModeButton.setToolTipText(searchDisplayMode.getToolTipText());
+    }
+
+    public void setAutoCompleter(AutoCompleter<String> searchCompleter) {
+        this.autoCompleteSupport.setAutoCompleter(searchCompleter);
+    }
+
+    public void performSearch() {
+        BasePanel currentBasePanel = getCurrentBasePanel();
+        if (currentBasePanel == null) {
+            return;
+        }
+
+        if (searchWorker != null) {
+            searchWorker.cancel(true);
+        }
+
+        // An empty search field should cause the search to be cleared.
+        if (searchField.getText().isEmpty()) {
+            clearSearch(currentBasePanel);
+            return;
+        }
+
+        SearchQuery searchQuery = getSearchQuery();
+        LOGGER.debug("Searching " + searchQuery + " in " + currentBasePanel.getTabTitle());
```

Comment:
`LOGGER.debug()` should be used to log error messages, here it seems more like a status update.